### ISSUE TITLE
TransactionalMap containsKey() blocks when key was previosuly locked by getForUpdate()

### ISSUE THREAD

**Issue by dmitrymz** [hazelcast/hazelcast#7588]
I wanted to check for object existence before calling getForUpdate().
Reason: hazelcast lockAndGet() (invoked from getForUpdate()) throws too generic TransactionException("Transaction couldn't obtain lock for the key") in case of failing to lock non-existing object, which is confusing, and hard to distinguish from important TransactionExceptions

```
    private V getByKey(TransactionalMap<String, V> region, String key) {
        return region.containsKey(key) ? region.getForUpdate(key) : null;
    }
```
however, the problem is that if the key is already locked by previous getForUpdate() then 
containsKey() method blocks.

I think this is erroneous behavior as containsKey() is a read-only operation that should not be blocked.
It doesn't matter if its result comes from inside or outside of transaction.

Please, advise of what is the right way to check for object existence before getForUpdate().
(Of course it would be great for lockAndGet to return some meaningful exception in case of object not found)

tried on v3.5.5

**Comment by gurbuzali** [comment]
Hi,
`lockAndGet` does not throw exception if the object(key) does not exists. That exception you are seeing (TransactionException("Transaction couldn't obtain lock for the key")) is a serious exception, it means somehow hazelcast couldn't locked the key (probably it is already locked by another transaction).

So you do not need to check for existence of an object. 

Regarding the blocking behavior of read-only operations in transactions, it is there by design to accomplish the atomicity of the transactions. We are planning to redesign this in 3.7.

Thank you

**Comment by jerrinot** [comment]
fixed by #7483.
